pkg/build/android.go

```diff
@@ -65,33 +62,151 @@ func (a android) readCompiler(archivePath string) (string, error) {
+	if err != nil {
+		return fmt.Errorf("failed to create command to install modules: %v", err)
+	}
+	cmd.Dir = params.KernelDir
+	cmd.Env = append([]string{}, moduleEnvVar)
+	if err := a.runCmd(cmd, params.KernelDir); err != nil {
+		return fmt.Errorf("failed to install modules: %v", err)
+	}
+
+	return nil
+}
+
 func (a android) build(params Params) (ImageDetails, error) {
 	var details ImageDetails
+	var err error
 
 	if params.CmdlineFile != "" {
 		return details, fmt.Errorf("cmdline file is not supported for android cuttlefish images")
 	}
 	if params.SysctlFile != "" {
 		return details, fmt.Errorf("sysctl file is not supported for android cuttlefish images")
 	}
+	commonKernelDir := filepath.Join(params.KernelDir, "common")
 
-	if err := a.runBuild(params.KernelDir, kernelConfig); err != nil {
-		return details, fmt.Errorf("failed to build kernel: %s", err)
+	// Build common kernel.
+	if err := a.buildCommonKernel(params.Config, params); err != nil {
+		return details, fmt.Errorf("failed to build android common kernel: %v", err)
+	}
+	if err := osutil.CopyFile(filepath.Join(params.OutputDir, "kernel.config"),
+		filepath.Join(params.OutputDir, "common-kernel.config")); err != nil {
+		return details, fmt.Errorf("failed to copy kernel config file: %v", err)
 	}
-	if err := a.runBuild(params.KernelDir, moduleConfig); err != nil {
-		return details, fmt.Errorf("failed to build modules: %s", err)
+
+	// Build external modules.
+	if err := a.buildExtModules(params); err != nil {
+		return details, fmt.Errorf("failed to build external modules: %v", err)
 	}
 
-	buildDistDir := filepath.Join(params.KernelDir, "dist")
-	bzImage := filepath.Join(buildDistDir, "bzImage")
-	vmlinux := filepath.Join(buildDistDir, "vmlinux")
-	initramfs := filepath.Join(buildDistDir, "initramfs.img")
+	// Zip kernel headers.
+	execHeadersScript := fmt.Sprintf("./%v", headersScript)
+	cmd := osutil.Command(execHeadersScript, commonKernelDir)
+	if err := a.runCmd(cmd, params.KernelDir); err != nil {
+		return details, fmt.Errorf("failed to zip kernel headers: %v", err)
+	}
 
-	buildOutDir := filepath.Join(params.KernelDir, "out")
-	config := filepath.Join(buildOutDir, "common", ".config")
+	// Create initramfs image.
+	execInitramfsScript := fmt.Sprintf("./%v", initramfsScript)
+	cmd = osutil.Command(execInitramfsScript, commonKernelDir)
+	if err := a.runCmd(cmd, params.KernelDir); err != nil {
+		return details, fmt.Errorf("failed to create initramfs image: %v", err)
+	}
 
-	var err error
-	details.CompilerID, err = a.readCompiler(filepath.Join(buildOutDir, "kernel-headers.tar.gz"))
+	bzImage := filepath.Join(commonKernelDir, "arch", "x86", "boot", "bzImage")
+	vmlinux := filepath.Join(commonKernelDir, "vmlinux")
+	initramfs := filepath.Join(commonKernelDir, "initramfs.img")
+
+	details.CompilerID, err = a.readCompiler(filepath.Join(commonKernelDir, "kernel-headers.tar.gz"))
```

Comment:
I think we can do
```
details.CompilerID, err := ...
```
here and drop the
```
var err error
```
above.

Reply:
That makes sense, but I'm receiving an error when I attempt to declare details.CompilerID

![image](https://user-images.githubusercontent.com/33524786/204405109-dd4e118d-1b64-4c6d-951f-2a3ff5b39d99.png)